avro2py/codegen.py

```diff
@@ -408,8 +409,29 @@ def visit_AnnAssign(self, node: ast.AnnAssign) -> ast.AnnAssign:
             return rewriter.visit(node)
 
 
+def resolve_schema_reference(schema: Record, schemas : List[Record]):
+    """Substitute the schema reference with the definition.""" 
+    if not schemas or schema.fully_qualified_name() in RESOLVED_SCHEMAS:
+        return
+
+    for field in schema.resolved_schema['fields']:
+        if isinstance(field['type'], str) and '.' in field['type']:
+            is_resolved = False
+            for s in schemas:
+                if field['type'] == s.fully_qualified_name():
+                    resolve_schema_reference(s, schemas)
+                    field['type'] = s.resolved_schema
+                    is_resolved = True
+                    break
+            if not is_resolved:
+                raise ValueError(f'Unable to resolve schema reference {field["type"]} in {schema.namespace}.{schema.name}')
```

Comment:
Don't use this pattern with the sentinel value - use instead:
```suggestion
            for s in schemas:
                if field['type'] == s.fully_qualified_name():
                    resolve_schema_reference(s, schemas)
                    field['type'] = s.resolved_schema
                    break
            else:
                raise ValueError(f'Unable to resolve schema reference {field["type"]} in {schema.namespace}.{schema.name}')
```
See https://docs.python.org/3/reference/compound_stmts.html#for

Reply:
I didn't know `else` can be used with `for`. Thanks for the suggestion.